libr/core/file.c

```diff
@@ -353,11 +353,22 @@ static int r_core_file_do_load_for_debug(RCore *r, ut64 baseaddr, const char *fi
 	}
 #endif
 	int fd = cf ? cf->fd : -1;
-	if (!r_bin_load (r->bin, filenameuri, baseaddr, UT64_MAX, xtr_idx, fd, treat_as_rawstr)) {
+
+	bo = r_bin_options_new (0LL, baseaddr, false);
+	if (!bo) {
+		eprintf ("Failed to create bin options\n");
+		return false;
+	}
+
+	bo->loadaddr = UT64_MAX;
+	bo->xtr_idx = xtr_idx;
+	bo->iofd = fd;
+	if (r_bin_open (r->bin, filenameuri, bo) < 0) {
```

Comment:
maybe we should check against -1 or have a constant for that like its done in libc-land?